lib/ui/home/home_view_model.dart

```
@@ -1,32 +1,102 @@
+    (ref) => ref.watch(homeViewModelProvider.notifier).today);
+
+final profileAvatarProvider = StreamProvider.autoDispose(
+    (ref) => ref.watch(homeViewModelProvider.notifier).profileAvatar);
+
+final surveysProvider = StreamProvider.autoDispose(
+    (ref) => ref.watch(homeViewModelProvider.notifier).surveys);
+
 class HomeViewModel extends StateNotifier<HomeViewState> {
   final GetProfileUseCase _getProfileUseCase;
+  final GetSurveysUseCase _getSurveysUseCase;
+
+  HomeViewModel(
+    this._getProfileUseCase,
+    this._getSurveysUseCase,
+  ) : super(const HomeViewState.init());
+
+  final BehaviorSubject<bool> _isLoading = BehaviorSubject();
+
+  Stream<bool> get isLoading => _isLoading.stream;
 
-  HomeViewModel(this._getProfileUseCase) : super(const HomeViewState.init());
+  final BehaviorSubject<String> _today = BehaviorSubject();
 
-  Future getProfile() async {
-    state = const HomeViewState.loading();
+  Stream<String> get today => _today.stream;
 
-    final result = await _getProfileUseCase.call();
+  final BehaviorSubject<String> _profileAvatar = BehaviorSubject();
+
+  Stream<String> get profileAvatar => _profileAvatar.stream;
+
+  final BehaviorSubject<List<SurveyModel>> _surveys = BehaviorSubject();
+
+  Stream<List<SurveyModel>> get surveys => _surveys.stream;
+
+  void loadData() async {
+    final profileStream = Stream.fromFuture(_getProfileUseCase.call());
+    final surveysStream = Stream.fromFuture(_getSurveysUseCase.call(
+      GetSurveysInput(
+        pageNumber: _defaultFirstPageIndex,
+        pageSize: _defaultPageSize,
+      ),
+    ));
+
+    ZipStream.zip2(
+      profileStream,
+      surveysStream,
+      (profileResult, surveysResult) {
+        _today.add(DateTime.now().getFormattedString());
+
+        if (profileResult is Success<ProfileModel>) {
+          _profileAvatar.add(profileResult.value.avatarUrl);
+        } else {
+          _profileAvatar.add('');
+        }
```

Should we skip emitting empty event? 🤔

---

As we ignore the error when fetch `profile` failed, I think we can @hoangnguyen92dn 👍